src/settings/ViewPrintDetails/ViewPrintDetails.js

```diff
@@ -0,0 +1,126 @@
+
+import { Loading, Modal, Button, ModalFooter } from '@folio/stripes/components';
+import { useCallout } from '@folio/stripes/core';
+
+import { usePrintDetailsSettings } from './hooks/usePrintDetailsSettings/usePrintDetailsSettings';
+import { usePrintDetailsSettingsMutation } from './hooks/usePrintDetailsSettingsMutation/usePrintDetailsSettingsMutation';
+import ViewPrintDetailsForm from './ViewPrintDetailsForm';
+import { VIEW_PRINT_DETAILS_ENABLED, PRINT_DETAILS_SETTINGS_KEY } from '../../constants';
+
+const ViewPrintDetails = () => {
+  const {
+    printDetailsInfo,
+    enablePrintLog,
+    isLoading,
+    refetch,
+  } = usePrintDetailsSettings();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [reRenderFlag, setReRenderFlag] = useState(true);
+  const { createPrintDetailsSettings, updatePrintDetailsSettings } = usePrintDetailsSettingsMutation();
+  const { formatMessage } = useIntl();
+  const callout = useCallout();
+
+  const createSetting = (values) => {
+    return createPrintDetailsSettings({
+      name: PRINT_DETAILS_SETTINGS_KEY,
+      value: {
+        enablePrintLog: values[VIEW_PRINT_DETAILS_ENABLED].toString(),
+      }
+    });
+  };
+
+  const updateSetting = (values) => {
+    return updatePrintDetailsSettings({
+      ...printDetailsInfo,
+      value: {
+        ...printDetailsInfo.value,
+        enablePrintLog: values[VIEW_PRINT_DETAILS_ENABLED].toString(),
+      }
+    });
+  };
+
+  const apiHandler = (values) => {
+    const handler = printDetailsInfo?.id ? updateSetting : createSetting;
+
+    return handler(values)
+      .then(() => {
+        refetch();
+        callout.sendCallout({
+          type: 'success',
+          message: formatMessage(
+            { id: 'ui-circulation.settings.ViewPrintDetails.callout.success' },
+          ),
+        });
+      })
+      .catch(() => {
+        callout.sendCallout({
+          type: 'error',
+          message: formatMessage(
+            { id: 'ui-circulation.settings.ViewPrintDetails.callout.error' },
+          ),
+        });
+      });
+  };
+
+  const onSubmit = async (values) => {
+    if (!values[VIEW_PRINT_DETAILS_ENABLED]) {
+      setIsModalOpen(true);
+    } else {
+      apiHandler(values);
+    }
+  };
+
+  const handleModalClose = () => {
+    setReRenderFlag(!reRenderFlag);
+    setIsModalOpen(false);
+  };
+
+  const handleModalYes = () => {
+    apiHandler({ [VIEW_PRINT_DETAILS_ENABLED]: false });
+    setIsModalOpen(false);
+  };
+
+  const modalFooter = (
+    <ModalFooter>
+      <Button
+        onClick={handleModalYes}
+        buttonStyle="primary"
+      >
+        {formatMessage({ id: 'ui-circulation.settings.ViewPrintDetails.warningPopupMessage.yes' })}
+      </Button>
+      <Button
+        onClick={handleModalClose}
+        buttonStyle="primary"
+      >
+        {formatMessage({ id: 'ui-circulation.settings.ViewPrintDetails.warningPopupMessage.no' })}
+      </Button>
+    </ModalFooter>
```

Comment:
If Confirmation Modal component is used instead of Modal component, this piece of implementation is intrinsically handled!